R/magpie-class.R

```diff
@@ -365,26 +356,25 @@ setMethod("[<-", #nolint
       if (is.factor(k)) k <- as.character(k)
       if (is.character(k) || is.list(k)) k <- .dimextract(x, k, 3, pmatch = pmatch)
     }
-    if (missing(value)) {
-      x@.Data[i] <- k
-      return(x)
-    } else {
-      if (is.magpie(value)) {
-        if (missing(i)) ii <- seq_len(dim(x)[1]) else ii <- i
-        if (missing(j)) jj <- seq_len(dim(x)[2]) else jj <- j
-        if (missing(k)) kk <- seq_len(dim(x)[3]) else kk <- k
-        value <- magpie_expand(value, x[ii, jj, kk])
-      } else if (length(value) != length(x@.Data[i, j, k]) & length(value) != 1) {
-        # dangerous writing of value as order might be wrong!
-        stop("Replacement does not work! Different replacement length!")
-      } else if (length(value) != 1 && !is.null(getOption("magclass.verbosity")) &&
-                 getOption("magclass.verbosity") > 1) {
-        message("NOTE ([<-): Dangerous replacement! As replacement value is not",
-          " an MAgPIE object name checking is deactivated!\n")
-      }
-      x@.Data[i, j, k] <- value
+    if (length(sys.call(which = -1)) == 4) {
+      x@.Data[i] <- value
       return(x)
     }
+    if (is.magpie(value)) {
+      if (missing(i)) ii <- seq_len(dim(x)[1]) else ii <- i
+      if (missing(j)) jj <- seq_len(dim(x)[2]) else jj <- j
+      if (missing(k)) kk <- seq_len(dim(x)[3]) else kk <- k
+      value <- magpie_expand(value, x[ii, jj, kk])
+    } else if (length(value) != length(x@.Data[i, j, k]) & length(value) != 1) {
```

Comment:
Did the linter not flag this? I thought `&` is always a linter warning in `if` statements
```suggestion
    } else if (length(value) != length(x@.Data[i, j, k]) && length(value) != 1) {
```